frontend/src/scalars/ui/Chart.vue

```diff
@@ -0,0 +1,598 @@
+<template>
+    <div class="visual-dl-page-charts">
+        <div class="visual-dl-chart-box" :style="computedStyle">
+        </div>
+        <div class="visual-dl-chart-actions">
+            <v-btn @click="expandArea">
+                <v-icon size="20">settings_overscan</v-icon>
+            </v-btn>
+            <v-select
+                v-if="downloadLink"
+                style="width:100px;"
+                hint="download type"
+                :items="runsItems"
+                v-model="downloadType"
+            />
+            <v-btn
+                v-if="downloadLink"
+                @click="handleDownLoad">
+                <v-icon size="20">file_download</v-icon>
+            </v-btn>
+        </div>
+    </div>
+</template>
+<script>
+import('vuetify/dist/vuetify.min.css')
+
+// TODO: Consider create Vue Components for these
+
+// // components
+// import Button from 'san-mui/Button';
+// import Icon from 'san-mui/Icon';
+// import DropDownMenu from '../../common/component/DropDownMenu';
+
+// libs
+import echarts from 'echarts';
+import axios from 'axios';
+import {isFinite, flatten, maxBy, minBy, sortBy, max, min} from 'lodash';
+import {generateJsonAndDownload} from '../../common/util/downLoadFile';
+import {quantile} from '../../common/util/index';
+import moment from 'moment';
+
+// service
+import {getPluginScalarsScalars} from '../../service';
+
+const originLinesOpacity = 0.3;
+const defaultSmoothWeight = 0.6;
+const lineWidth = 1.5;
+const minQuantile = 0.05;
+const maxQuantile = 0.95;
+// the time to refresh chart data
+const intervalTime = 15;
+
+export default {
+    components: {
+        // 'ui-dropdown-menu': DropDownMenu,
+        // 'san-button': Button,
+        // 'san-icon': Icon
+    },
+    props: ['tagInfo', 'groupNameReg', 'smoothing', 'horizontal', 'sortingMethod', 'downloadLink', 'outlier', 'runs',
+            'running', 'runsItems'],
+    computed: {
+        computedStyle() {
+            return 'height:' + this.height + 'px;'
+                + 'width:' + this.width + 'px;';
+        }
+    },
+    data() {
+        return {
+            width: 400,
```

Comment:
Why are these stored as component data? Unless this changes (which we don't have right now right?), this should probably not be computed

Reply:
Storing width and height in the data seems fine.

Reply:
I think they put it here because user can expand/shrink the chart